### ISSUE TITLE
a problem about llama_cpp python

### ISSUE THREAD

**Issue by linuarM** [SeargeDP/ComfyUI_Searge_LLM#28]
### so i am getting a error of the llama_cpp. 
 Llama = importlib.import_module("llama_cpp").Llama
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
### i tried running this commands

python -m pip install https://github.com/oobabooga/llama-cpp-python-cuBLAS-wheels/releases/download/cpu/llama_cpp_python-0.2.89+cpuavx2-cp311-cp311-win_amd64.whl
python -m pip install https://github.com/oobabooga/llama-cpp-python-cuBLAS-wheels/releases/download/textgen-webui/llama_cpp_python_cuda-0.2.89+cu121-cp311-cp311-win_amd64.whl

###  but i got error of 

uavx2-cp311-cp311-win_amd64.whl
ERROR: llama_cpp_python-0.2.89+cpuavx2-cp311-cp311-win_amd64.whl is not a supported wheel on this platform.

C:\ComfyUI_windows_portable_nvidia 002\ComfyUI_windows_portable\python_embeded>python -m pip install https://github.com/oobabooga/llama-cpp-python-cuBLAS-wheels/releases/download/textgen-webui/llama_cpp_python_cuda-0.2.89+cu121-cp311-cp311-win_amd64.whl
ERROR: llama_cpp_python_cuda-0.2.89+cu121-cp311-cp311-win_amd64.whl is not a supported wheel on this platform.

### so i tried a cpu and cuda 124 version. which is my cuda version. it gave this error.

C:\ComfyUI_windows_portable_nvidia 002\ComfyUI_windows_portable\python_embeded>python -m pip install https://github.com/oobabooga/llama-cpp-python-cuBLAS-wheels/releases/download/cpu/llama_cpp_python-0.2.89+cpuavx2-cp311-cp311-win_amd64.whl
ERROR: llama_cpp_python-0.2.89+cpuavx2-cp311-cp311-win_amd64.whl is not a supported wheel on this platform.

C:\ComfyUI_windows_portable_nvidia 002\ComfyUI_windows_portable\python_embeded>python -m pip install torch==2.4.1+cu124 torchvision torchaudio --index-url https://github.com/oobabooga/llama-cpp-python-cuBLAS-wheels/releases/download/cpu/llama_cpp_python-0.2.89+cpuavx2-cp311-cp311-win_amd64.whl
Looking in indexes: https://github.com/oobabooga/llama-cpp-python-cuBLAS-wheels/releases/download/cpu/llama_cpp_python-0.2.89+cpuavx2-cp311-cp311-win_amd64.whl

ERROR: Could not find a version that satisfies the requirement torch==2.4.1+cu124 (from versions: none)
ERROR: No matching distribution found for torch==2.4.1+cu124

(links showing only cpu commands but i did cuda versions too error is same as well)

should i change my python version? it is Python 3.10.6

**Comment by linuarM** [comment]
> [linuarM](/linuarM)

im guessing the issue is torch==2.4.1+cu124 is not a thing i can download in pytorch. so maybe 2.4.1+cu121 or else could work. but dont know if it will cause to another gpu related problems when ran..

**Comment by AliHamandy** [comment]
same problem here

**Comment by trollishka** [comment]
yup. same

**Comment by MarekWo** [comment]
Same problem, so I was wondering if there was a solution for that, but apparently there isn't yet.

**Comment by linuarM** [comment]
@MarekWo @trollishka @AliHamandy did you guys saw this one: https://github.com/SeargeDP/ComfyUI_Searge_LLM/issues/3 ? in there they first check if theres llama installed. then install it manually in the environment. im confused about the part when they say that installing manually in python environment. because i dont know how do i going to know that i am in the environment or not
.

**Comment by AliHamandy** [comment]
I think the versions that can be installed manually you python 3.11 and less so if you're using python 3.12 you'll need to downgrade to python 3.11 to find compatibility and it will work 
Or just don't use this one since there are way better ways to make text2text or image2text in comfyui like the ollama models they are up to date and really good consider trying them

**Comment by linuarM** [comment]
@AliHamandy you are right. since i didnt tried upgrading my python yet. i will public my results when i can. i'll also going to check ollama models, thank you very much for your recomendation.

**Comment by AI421** [comment]
python 3.12 have issue to install llama_cpp_cuda, when  it run on CPU, it's extremely slow

**Comment by jmorbit** [comment]
I was having the same problem getting this to work, constant import failed until I dug deeper into the log and saw that CMAKE was not installed. I went to https://cmake.org/download/ and installed the appropriate CMAKE, then I had to go to https://pypi.org/project/llama-cpp-python/ and "pip install llama-cpp-python". After installing CMAKE, llama-cpp installed with no issues and the import of the node  worked.
